cpp/src/io/json/json_gpu.cu

```diff
@@ -346,29 +344,28 @@ struct ConvertFunctor {
  * @param[in] ch The character to check
  *
  * @return True if the input is whitespace, False otherwise
- **/
+ */
 __inline__ __device__ bool is_whitespace(char ch) { return ch == '\t' || ch == ' '; }
 
 /**
- * @brief Scans a character stream within a range, and adjusts the start and end
- * indices of the range to ignore whitespace and quotation characters.
+ * @brief Adjusts the range to ignore starting/trailing whitespace and quotation characters.
  *
- * @param[in] data The character stream to scan
- * @param[in,out] start The start index to adjust
- * @param[in,out] end The end index to adjust
- * @param[in] quotechar The character used to denote quotes
+ * @param[in] begin Pointer to the first character in the parsing range
+ * @param[in] end pointer to the first character after the parsing range
+ * @param[in] quotechar The character used to denote quotes; '\0' if none
  *
- * @return Adjusted or unchanged start_idx and end_idx
- **/
-__inline__ __device__ void trim_field_start_end(const char *data,
-                                                long *start,
-                                                long *end,
-                                                char quotechar = '\0')
+ * @return Trimmed range
+ */
+__inline__ __device__ std::pair<char const *, char const *> trim_whitespaces_quotes(
+  char const *begin, char const *end, char quotechar = '\0')
 {
-  while ((*start < *end) && is_whitespace(data[*start])) { (*start)++; }
-  if ((*start < *end) && data[*start] == quotechar) { (*start)++; }
-  while ((*start <= *end) && is_whitespace(data[*end])) { (*end)--; }
-  if ((*start <= *end) && data[*end] == quotechar) { (*end)--; }
+  auto first = begin;
+  auto last  = end - 1;
+  while ((first < last) && is_whitespace(*first)) { first++; }
+  if ((first < last) && *first == quotechar) { first++; }
+  while ((first <= last) && is_whitespace(*last)) { last--; }
+  if ((first <= last) && *last == quotechar) { last--; }
+  return {first, last + 1};
```

Comment:
Update to use `thrust::find_if` & `thrust::find_if` + `thrust::make_reverse_iterator`